src/components/user/edit/basic-details.vue

```diff
@@ -46,12 +49,16 @@ defineOptions({
 
 // The component assumes that this data will exist when the component is
 // created.
-const { user } = useRequestData();
+const { currentUser, user } = useRequestData();
 const { awaitingResponse } = user.toRefs();
 
 const email = ref(user.email);
 const displayName = ref(user.displayName);
 
+const config = inject('config');
+const emailDisabled = computed(() =>
+  config.oidcEnabled && !currentUser.can('user.update'));
```

Comment:
email should be disabled whenever `config.oidcEnabled` **or** when current user can't update user

Reply:
I thought about it some more, but I really think it's "and", not "or"!

- If OIDC is not enabled, and the user is updating their own email on /account/edit, that is allowed. We don't need to check for `user.update` or disable the input.
- If OIDC is not enabled, and the user is updating someone's email on /users/:id/edit, then the `validateData` route meta field checks that the user can `user.update`. That means that if the user is able to see the input at all, we know they can use it, so it doesn't need to be separately disabled.
- If OIDC is enabled, and the user wants to update their own email on /account/edit, that should only be allowed if they are an admin (if they can `user.update`). I mentioned in a comment above that admins should still be able to update email addresses. In other words, the input should be disabled if OIDC is enabled and the user can't `user.update`.
- If OIDC is enabled, and the user wants to update someone's email on /users/:id/edit, the `validateData` meta field will again check for `user.update`.